### PR TITLE
[Embeds] Embed for single post

### PR DESCRIPTION
**Pull request by mozzius**
# Stacked on #3448 

This PR adds support for embedding a single post. It takes an AT URI in the search parameters, and supports rich text and embeds

Screenshot 2024-04-09 at 00 00 55

## Test plan

1. Start the dev server (`cd bskyembed` and `yarn dev`)
2. Find the AT URI of a post you want to test (I do this by opening the network tab, clicking on a post, then searching for getPostThread - it's passed as a query string)
3. Add `?uri=<at-uri>` to the query string (i.e. `http://localhost:5173/?uri=at://did:plc:wqwdm7xgynf7amclnjfhvncr/app.bsky.feed.post/3kpvyeesjry2o`)
4. Test all kinds of posts you can think of - different combinations of embeds, complicated facets etc

### CODE REVIEW

**Comment by mozzius** [comment]
Image embeds

Screenshot 2024-04-09 at 14 09 45
Screenshot 2024-04-09 at 14 09 29
Screenshot 2024-04-09 at 14 09 01
Screenshot 2024-04-09 at 14 08 36
Screenshot 2024-04-09 at 14 08 18

**Comment by mozzius** [comment]
external embed

Screenshot 2024-04-09 at 14 28 03

**Comment by mozzius** [comment]
Quote post (+ with media)

Screenshot 2024-04-09 at 14 46 48
Screenshot 2024-04-09 at 14 46 09
Screenshot 2024-04-09 at 14 42 19

**Comment by mozzius** [comment]
Lists and feeds
Screenshot 2024-04-09 at 15 19 38
Screenshot 2024-04-09 at 15 12 42

**Comment by mozzius** [comment]
PWI opt out

Screenshot 2024-04-09 at 16 49 23
Screenshot 2024-04-09 at 16 49 17

File: bskyweb/static.go [hunk 1]
```diff
@@ -2,5 +2,5 @@ package bskyweb
 
 import "embed"
 
-//go:embed static/*
+//go:embed all:static/*
```
Comment (mozzius) [inline bskyweb/static.go:5]: I'm not really familiar with what's going on here, but CI was failing - relevant issue here:

https://github.com/golang/go/issues/43854

File: bskyembed/src/embed.tsx [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+      }
+
+      // Case 3.5: Post not found
+      if (AppBskyEmbedRecord.isViewNotFound(record)) {
+        return <Info>Quoted post not found - it may have been deleted</Info>
+      }
+
+      // Case 3.6: Post blocked
+      if (AppBskyEmbedRecord.isViewBlocked(record)) {
+        return <Info>The quoted post is blocked</Info>
```
Comment (haileyok) [inline bskyembed/src/embed.tsx:133]: Do we want to give more context on what "blocked" means? Might be confusing for someone who doesn't use Bluesky, but idk.

Comment (mozzius) [reply bskyembed/src/embed.tsx:133]: Yeah, I went back and forth on this but I wasn't able to summarise the rather complex behaviour at play here 😅 open to suggestions

File: bskyembed/src/main.tsx [hunk 3]
```diff
@@ -1,9 +1,88 @@
...
+import {getRkey} from './utils'
 
 const root = document.getElementById('app')
 if (!root) throw new Error('No root element')
-render(<App />, root)
+
+const searchParams = new URLSearchParams(window.location.search)
+
+const agent = new BskyAgent({
+  service: searchParams.get('service') || 'https://public.api.bsky.app',
```
Comment (pfrazee) [inline bskyembed/src/main.tsx:18]: I don't think we can do an open proxy like this

Comment (pfrazee) [reply bskyembed/src/main.tsx:18]: even with the xrpc constraints we'd probably be smart to just hardcode this to our service

Comment (pfrazee) [reply bskyembed/src/main.tsx:18]: Oh my bad, this seems to be client side. In that case, not a concern

Comment (pfrazee) [reply bskyembed/src/main.tsx:18]: Right. Ignore me.

Comment (mozzius) [reply bskyembed/src/main.tsx]: Oh - just removed it. Should I revert?

File: bskyembed/src/embed.tsx (hunk 2 above)
Comment (haileyok) [reply bskyembed/src/embed.tsx:133]: Ha yea notice I didn't give a suggestion 😅 not really sure how to word this either based on the actual behavior but also trying to be relatively simple for outsiders.

File: bskyembed/src/embed.tsx [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+      // Case 3.1: Post
+      if (AppBskyEmbedRecord.isViewRecord(record)) {
+        const pwiOptOut = !!record.author.labels?.find(
+          label => label.val === '!no-unauthenticated',
+        )
+        if (pwiOptOut) {
+          return (
+            <Info>
+              The author of this quoted post has requested their posts not be
```
Comment (matthieusieben) [inline bskyembed/src/embed.tsx:43]: Can the locale be configured via the embed? Or is the locale of the post gonna be used?

Comment (mozzius) [reply bskyembed/src/embed.tsx:43]: Good question, I'll add to the trello

**Review by gaearon — APPROVED** [review]
amazing

**Pull request merged**